pkg/repositories/models/resource.go

```diff
@@ -0,0 +1,25 @@
+package models
+
+type ResourcePriority int32
+
+const (
+	ResourcePriorityDomainLevel        ResourcePriority = 1
+	ResourcePriorityProjectDomainLevel ResourcePriority = 10
+	ResourcePriorityWorkflowLevel      ResourcePriority = 100
+	ResourcePriorityLaunchPlanLevel    ResourcePriority = 1000
+)
+
+// Represents Flyte resources repository.
+// In this model, the combination of (Project, Domain, Workflow, LaunchPlan, ResourceType) is unique
+type Resource struct {
+	BaseModel
+	ID           int64  `gorm:"AUTO_INCREMENT;column:id;primary_key"`
```

Comment:
how does this field work with the BaseModel ID field? do the struct tags here still get applied?

Reply:
Oh yes. I don't believe I can use the BaseModel then. Should I just copy over 

```	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time `sql:"index"```

Also that is one of the things I wanted to ask - how can I do end to end test?